src/python/pants/base/exception_sink.py

```diff
@@ -111,8 +111,9 @@ class ExceptionSink:
   # to properly setup global state.
   _log_dir = None
   # We need an exiter in order to know what to do after we log a fatal exception or handle a
-  # catchable signal.
-  _exiter: Optional[Exiter] = None
+  # catchable signal. We default to `sys.exit` to configure the behavior for import time, but expect
+  # this to changed for later in the Pants bootstrap process.
+  _exiter: Exiter = Exiter(exiter=sys.exit)
```

Comment:
@cosmicexplorer can you please sanity check that it is safe to move the default from line 546 to this line? It greatly simplifies the type hints for all this `ExceptionSink/Exiter` related code because `ExceptionSink._exiter` no longer needs to be `Optional`.